lib/natalie/compiler2/pass1.rb

```diff
@@ -33,6 +33,21 @@ def transform_body(body, used:)
         instructions
       end
 
+      def constant_name(instructions, name)
+        if name.is_a?(Symbol)
+          instructions << PushSelfInstruction.new
+        elsif name.sexp_type == :colon2
+          _, namespace, name = name
+          instructions << transform_expression(namespace, used: true)
+        elsif name.sexp_type == :colon3
+          _, name = name
+          instructions << PushObjectClassInstruction.new
+        else
+          raise "Unknown constant name type #{name.sexp_type.inspect}"
+        end
+        name
```

Comment:
Let's return a pair (array) of `[instructions, name]` from this method rather than mutating the `instructions` array in place. That way it can be a pure function that has no side effects.